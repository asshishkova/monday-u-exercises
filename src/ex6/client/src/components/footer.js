import React, { useState } from "react";
import PropTypes from "prop-types";
import { sortItems, clearAllItems } from "../item-client.js";
import "../styles/footer.css";


export function Footer(props) {
  const {updateTodos, amount} = props;
  const [sortBy, setSortBy] = useState('Sort by');

  const onSortListButtonClicked = async () => {
    setSortBy(await sortItems());
    await updateTodos();
  }

  const onClearAllButtonClicked = async () => {
    if (window.confirm('Are you sure?')) {
      await clearAllItems();
      await updateTodos();
    }
  }

  let tasks = "tasks";
  if (amount === 1) tasks = "task";
  const amountInfo = <p id="amount-info">{amount} pending {tasks}</p>
  const clearAllButton = <button id="clear-all-button" className="btn" onClick={onClearAllButtonClicked}>Clear all</button>
  const sortListButton =  <button id="sort-list-button" className="btn" onClick={onSortListButtonClicked}>
                            {sortBy} <i className="fa fa-caret-down"></i>
                          </button>

  return (
    <footer id="footer">
      { amount > 1 && sortListButton }
      { amount > 0 && amountInfo }
      { amount > 1 && clearAllButton }
    </footer>
  )
}

Footer.propTypes = {
  updateTodos: PropTypes.func,
  amount: PropTypes.number
}
