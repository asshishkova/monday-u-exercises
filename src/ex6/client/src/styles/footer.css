#footer {
  display: flex;
  justify-content: center;
}

#sort-list-button {
  background-color: var(--light-green);
  color: var(--very-dark-green);
  width: 115px;
  display: flex;
  justify-content: space-between;
  padding: 1px 8px;
}

.fa-caret-down {
  margin-top: 5px;
}

#footer p {
  margin: 0;
  width: 230px;
  padding: auto;
  text-align: center;
}

#amount-info {
  margin: auto;
}

#clear-all-button {
  background-color: var(--dark-green);
  width: 115px;
}
