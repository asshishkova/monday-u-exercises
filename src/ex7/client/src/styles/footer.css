#footer {
  display: flex;
  justify-content: space-between;
}

#amount-info, #restore-deleted {
  text-align: left;
  margin: 0px 0px 0px 10px;
}

#clear-all-button {
  background-color: var(--dark-green);
  width: 115px;
}

#restore-deleted {
  cursor: pointer;
  color: brown;
}
