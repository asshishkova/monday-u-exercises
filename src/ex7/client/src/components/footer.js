import React from "react";
import PropTypes from "prop-types";
import { clearAllItems } from "../item-client.js";
import "../styles/footer.css";


export function Footer(props) {
  const {updateTodos, todos} = props;
  const amount = todos.length;
  const amountDone = todos.filter((todo) => todo.status).length;
  const amountPending = amount - amountDone;

  const onClearAllButtonClicked = async () => {
    if (window.confirm('Are you sure?')) {
      await clearAllItems();
      await updateTodos();
    }
  }

  const amountInfo = <p id="amount-info">Tasks: {amountPending} pending, {amountDone} done.</p>
  const clearAllButton = <button id="clear-all-button" className="btn" onClick={onClearAllButtonClicked}>Clear all</button>

  return (
    <footer id="footer">
      { amount > 0 && amountInfo }
      { amount > 0 && clearAllButton }
    </footer>
  )
}

Footer.propTypes = {
  updateTodos: PropTypes.func,
  todos: PropTypes.array
}
