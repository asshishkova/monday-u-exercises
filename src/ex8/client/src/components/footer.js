import React from "react";
import { clearAllItems, restoreItem } from "../item-client.js";
import "../styles/footer.css";

export function Footer({  lastDeletedItem, todos,
                          setServerErrorMessageAction,
                          clearServerErrorMessageAction,
                          saveDeletedItemAction,
                          setTodosAction,
                          addTodosAction }) {

  const amount = todos.length;
  const amountDone = todos.filter((todo) => todo.status).length;
  const amountPending = amount - amountDone;

  const onClearAllButtonClicked = async () => {
    clearServerErrorMessageAction();
    saveDeletedItemAction(null);
    if (window.confirm('Are you sure?')) {
      try {
        await clearAllItems();
        setTodosAction([]);
      } catch (error) {
        setServerErrorMessageAction(`Error: ${error.message}`)
      }
    }
  }

  const restoreDeletedTodo = async () => {
    clearServerErrorMessageAction();
    try {
      await restoreItem(lastDeletedItem);
      saveDeletedItemAction(null);
      addTodosAction([lastDeletedItem]);
    } catch (error) {
      setServerErrorMessageAction(`Error: ${error.message}`)
    }
  }

  const amountInfo = <p id="amount-info">Tasks: {amountPending} pending, {amountDone} done.</p>;
  const clearAllButton =  <button id="clear-all-button" className="btn"
                            onClick={onClearAllButtonClicked}>Clear all
                          </button>;
  const restoreDeletedIcon =  <button id="restore-deleted" className="btn"
                                onClick={restoreDeletedTodo}><i className="fa fa-undo" aria-hidden="true"></i>
                              </button>
  return (
    <div>
      { amount > 0 &&
        <footer id="footer">
          { amountInfo }
          { lastDeletedItem && restoreDeletedIcon }
          { clearAllButton }
        </footer>
      }
    </div>
  )
}
