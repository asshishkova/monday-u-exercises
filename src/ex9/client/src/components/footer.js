import React from "react";
import "../styles/footer.css";

export function Footer({  lastDeletedItem, todos,
                          setServerErrorMessageAction,
                          clearServerErrorMessageAction,
                          saveDeletedItemAction,
                          clearTodosAction,
                          restoreTodoAction }) {

  const amount = todos.length;
  const amountDone = todos.filter((todo) => todo.status).length;
  const amountPending = amount - amountDone;

  const onClearAllButtonClicked = async () => {
    clearServerErrorMessageAction();
    saveDeletedItemAction(null);
    if (window.confirm('Are you sure?')) {
      try {
        await clearTodosAction();
      } catch (error) {
        setServerErrorMessageAction(`Error: ${error.message}`)
      }
    }
  }

  const restoreDeletedTodo = async () => {
    clearServerErrorMessageAction();
    try {
      saveDeletedItemAction(null);
      await restoreTodoAction(lastDeletedItem);
    } catch (error) {
      setServerErrorMessageAction(`Error: ${error.message}`)
    }
  }

  const amountInfo = <p id="amount-info">Tasks: {amountPending} pending, {amountDone} done.</p>;
  const clearAllButton =  <button id="clear-all-button" className="btn"
                            onClick={onClearAllButtonClicked}>Clear all
                          </button>;
  const restoreDeletedIcon =  <button id="restore-deleted" className="btn"
                                onClick={restoreDeletedTodo}><i className="fa fa-undo" aria-hidden="true"></i>
                              </button>
  return (
    <div>
      { amount > 0 &&
        <footer id="footer">
          { amountInfo }
          { lastDeletedItem && restoreDeletedIcon }
          { clearAllButton }
        </footer>
      }
    </div>
  )
}
