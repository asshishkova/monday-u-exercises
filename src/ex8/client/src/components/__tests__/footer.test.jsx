import renderer from "react-test-renderer";
import { Footer } from "../footer.js";

const todos = [
  {id: 1, text: 'one', status: false, done: null},
  {id: 2, text: 'two', status: true, done: "2022-07-13T12:49:57.000Z"},
  {id: 3, text: 'three', status: false, done: null}
]

const lastDeletedItem = {id: 4, text: 'four', status: false, done: null};

test('renders correctly amount info of todos without last deleted item', () => {
  const tree = renderer.create(
    <Footer lastDeletedItem={null} todos={todos}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly amount info of todos with last deleted item', () => {
  const tree = renderer.create(
    <Footer lastDeletedItem={lastDeletedItem} todos={todos}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
