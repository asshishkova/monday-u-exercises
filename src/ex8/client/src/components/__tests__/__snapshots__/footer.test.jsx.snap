// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly amount info of todos with last deleted item 1`] = `
<div>
  <footer
    id="footer"
  >
    <p
      id="amount-info"
    >
      Tasks: 
      2
       pending, 
      1
       done.
    </p>
    <button
      className="btn"
      id="restore-deleted"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-undo"
      />
    </button>
    <button
      className="btn"
      id="clear-all-button"
      onClick={[Function]}
    >
      Clear all
    </button>
  </footer>
</div>
`;

exports[`renders correctly amount info of todos without last deleted item 1`] = `
<div>
  <footer
    id="footer"
  >
    <p
      id="amount-info"
    >
      Tasks: 
      2
       pending, 
      1
       done.
    </p>
    <button
      className="btn"
      id="clear-all-button"
      onClick={[Function]}
    >
      Clear all
    </button>
  </footer>
</div>
`;
